tool/src/main/java/org/wildfly/security/tool/Command.java

```diff
@@ -270,4 +309,91 @@ protected Supplier<Provider[]> getProvidersSupplier(final String providersList)
             }
         };
     }
+
+    /**
+     * Acquire a given secret key from a {@link CredentialStore}.
+     *
+     * @param alias the name for a secret key within the CredentialStore
+     * @return the requested {@link SecretKey}, or {@code null} if it could not be retrieved
+     */
+    SecretKey getSecretKey(Boolean createCredentialStore, String credentialStoreLocation, String alias, Boolean populate,
+                           String inputRealmLocation) throws Exception {
+        CredentialStore credentialStore;
+        // check if credential-store and secret-key-alias are both specified, or both null
+        String csType = PropertiesCredentialStore.NAME;
+        try {
+            credentialStore = CredentialStore.getInstance(csType);
+        } catch (NoSuchAlgorithmException e) {
+            // fallback to load all possible providers
+            credentialStore = CredentialStore.getInstance(csType, getProvidersSupplier(null));
+        }
+        Map<String, String> implProps = new HashMap<>();
+        implProps.put("create", String.valueOf(createCredentialStore));
+        implProps.put("location", credentialStoreLocation);
+        implProps.put("modifiable", Boolean.TRUE.toString());
+        credentialStore.initialize(implProps);
+        try {
+            credentialStore.retrieve(alias, SecretKeyCredential.class).getSecretKey();
+            System.out.println(ElytronToolMessages.msg.existingCredentialStore());
+        } catch (Exception e) {
+            if (!createCredentialStore) {
+                warningHandler(ElytronToolMessages.msg.skippingBlockMissingCredentialStore());
+                return null;
+            }
+            if (populate) {
+                SecretKey key = SecretKeyUtil.generateSecretKey(256);
+                Credential keyCredential = new SecretKeyCredential(key);
+                credentialStore.store(alias, keyCredential);
+                credentialStore.flush();
+            } else {
+                errorHandler(ElytronToolMessages.msg.cmdFileSystemPopulateUnspecified());
+            }
+        }
+        SecretKey key;
+        try {
+            key = credentialStore.retrieve(alias, SecretKeyCredential.class).getSecretKey();
+        } catch (NullPointerException e) {
```

Comment:
Instead of catching the `NullPointerException` here, it would be better to check the return value of `credentialStore.retrieve` prior to calling `getSecretKey`.

Reply:
I'll update this and the other section to be similar to `getKeyPair`. This code was unchanged from the original encrypt command, so I'm not as familiar with the functionality.

Reply:
@OndrejKotek Noting that I made some changes in response to this, in the new 5th commit